Automattic/simplenote-ios
# Nü Note Options: Mark I

### 🛑  Important
Don't merge until #906 #907 #909 and #910 are ready: everything must be merged in a single shot!

### Details:

- Implements the new Note Options Interface. **None of its actions is wired at this stage!**
- New TableViewCells: **SwitchTableViewCell** + **Value1TableViewCell**
- Replaces the leagacy Action Sheet with the new Options UI
- Adds new Images!

@aerych Sir!! New UI!! May I bother you with this one?
**Thank you!!!**

Closes #22 
Closes #457
Ref. #763
Ref. #863 

### Test
1. Open any note
2. Press over the top `( i )` icon

- [x] Verify that pressing over the Switch rows (anywhere) flips the Switch
- [x] Verify that Text Only cells (with no switch) get immediately deselected (except for the Copy Link row!)
- [x] Verify this new UI looks great in Light / Dark mode
- [x] Verify this new UI responds well to Right to Left languages please!
- [x] Verify on iPad devices the new Options UI appears as a popover, and upon rotation remains anchored!

**Note:** Action Handlers are implemented in follow-ups!

### Release
These changes do not require release notes.

### Screenshots
<img width="200" src="https://user-images.githubusercontent.com/1195260/95612298-91c5b900-0a39-11eb-9dd4-85e959eb5ff5.png"> <img width="200" src="https://user-images.githubusercontent.com/1195260/95612311-95594000-0a39-11eb-8001-f0e248859d24.png">

<img width="300" src="https://user-images.githubusercontent.com/1195260/95612390-aefa8780-0a39-11eb-85c6-0b7fc9efdb6c.png"> <img width="300" src="https://user-images.githubusercontent.com/1195260/95612394-b02bb480-0a39-11eb-9271-888c91b5465b.png">
